src/components/BookList.jsx

```diff
@@ -1,10 +1,33 @@
-import { useSelector } from 'react-redux';
-
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getBooks } from '../api/bookstoreApi';
 import AddBook from './AddBook';
 import Book from './Book';
 
 function BookList() {
-  const books = useSelector((state) => state.books);
+  const {
+    books, length, isLoading, hasError, errorMessage,
+  } = useSelector((store) => store.books);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getBooks());
+  }, [dispatch, length]);
+
+  if (isLoading) {
+    return <div className="loading">Loading...</div>;
+  }
```

Comment:
- [OPTIONAL] You've implemented the loading request feature really well! Just a thought for further enhancing the user experience: consider minimizing disruptions by avoiding a full page reload when adding or removing a book. While your loading indicator effectively communicates ongoing processes, you can keep the existing books visible while the new book is being added or removed in the background. This approach provides a smoother experience, as users won't lose their context by witnessing the entire list disappear and reappear. Keep up the good work! 📚🚀

Reply:
Thanks you very much for the suggestion. I will implement changes as suggested